ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -65,12 +68,46 @@ public void connect(final String url, final int id) {
     // Disable timeouts for read
     client.setReadTimeout(0, TimeUnit.MINUTES);
 
-    Request request = new Request.Builder()
+    Request.Builder builder = new Request.Builder()
         .tag(id)
-        .url(url)
-        .build();
+        .url(url);
 
-    WebSocketCall.create(client, request).enqueue(new WebSocketListener() {
+    if (options != null) {
+      if (options.hasKey("origin")) {
+        builder.addHeader("Origin", options.getString("origin"));
+      }
+
+      if (options.hasKey("headers")) {
```

Comment:
The ws module's options argument separates "origin" from the other header values that can be added to the request. Rather than duplicate their api exactly, this seemed like a reasonable start to allow for request customization.